src/renderer/components/+namespaces/namespace.store.ts

```diff
@@ -106,8 +104,14 @@ export class NamespaceStore extends KubeObjectStore<Namespace> {
   getContextNamespaces(): string[] {
     const namespaces = this.contextNs.toJS();
 
+    // show all namespaces when nothing selected
     if (!namespaces.length) {
-      return [...this.allowedNamespaces]; // show all namespaces when nothing selected
+      if (this.isLoaded) {
+        // return actual namespaces list since "allowedNamespaces" updating every 30s in cluster and thus might be stale
+        return this.items.map(namespace => namespace.getName());
```

Comment:
Why is this more up to date? This seems to be following a bad precedent of multiple sources of truth.

Reply:
This is technical implementation. Consider a case:

1. You have "All namespaces" selected filter
2. New namespace created and added via watch-api to the store (but it's not included yet to `cluster.allowedNamespace` since it comes from main process and updating every 30s)
3. You're switching to other resource tab and don't see anything about that namespace